lib/lightning/credentials/credential.ex

```diff
@@ -50,6 +50,9 @@ defmodule Lightning.Credentials.Credential do
     |> unique_constraint([:name, :user_id],
       message: "you have another credential with the same name"
     )
+    |> unique_constraint([:external_id, :user_id],
+      message: "you have another credential with the external ID"
+    )
```

Comment:
This constraint enforces uniqueness per-user, but the bug we're fixing requires uniqueness per-project, right ? Also, this needs a corresponding database index/migration to actually work. I scanned a little bit the migrations and the changes in this PR but I haven't seen one. Without a db migration for the constraint, this probably won't work.